[v1.10.x] Add check for python in runfabtests.sh

shijin-aws opened the pull request:
This PR backports https://github.com/ofiwg/libfabric/pull/6052 and https://github.com/ofiwg/libfabric/pull/6054 to v1.10.x branch. It adds a check for the availability of python command (and its variants: python2, python3) before running tests. This will guarantee the error code will not mess up when the default `python` command is not available on the system.

shijin-aws commented:
@j-xiong is the failure of `continuous-integration/jenkins/pr-head` related to https://github.com/ofiwg/libfabric/issues/7027 or some other errors?

j-xiong commented:
@shijin-aws Here is the details of the failure:
```
13:26:11  git clone --depth 1 https://github.com/ParRes/ISx.git ISx
13:26:11  None
13:26:11  Cloning into 'ISx'...
13:28:32  fatal: unable to access 'https://github.com/ParRes/ISx.git/': Failed connect to github.com:443; Connection timed out
13:28:32  exiting with 128
```
Seems to be some network connection issue.

shijin-aws commented:
Can we re-trigger this test?

j-xiong commented:
@shijin-aws I tried twice. Still the same error. Need more investigation to find out the root cause.

shijin-aws commented:
@j-xiong Thanks. It will be appreciated if we can rerun the test after the root cause resolved.

aingerson commented:
The cloning issue showed up a while ago upstream. We don't know the root caused but @nikhilnanal removed the cloning altogether to avoid it but the fix was never backported to 1.10.x. See #7040

shijin-aws commented:
@j-xiong May I know what's the error in Jenkins CI? I already ingest the recently merged Jenkins CI fix in my updated PR.

```
commit 738805afc138771752d1b140e1fcd4d374d21c25 (upstream/v1.10.x)
Merge: c76b7c34d f69ac2862
Author: Jianxin Xiong <jianxin.xiong@intel.com>
Date:   Tue Sep 7 12:02:52 2021 -0700

    Merge pull request #7040 from nikhilnanal/v1.10.x

    CI/Jenkins: changed build code for shmem tests due to git clone issues

commit f69ac286229da6ab41f231ca8515f749e0b08a8f
Author: Nikhil Nanal <nikhil.nanal@intel.com>
Date:   Thu Jul 30 15:50:36 2020 -0700

    CI/Jenkins: changed build code for shmem tests due to git clone issues

    Changed the shmem build to source the ISx, PRK and uh from
    prefetched repositories instead of attempting to clone run time.
    This is done since shmem build is throwing https errors while
    cloning these repos on the fly. This also saves build time.

    cherry-picked from main 7257067798e52dd57451594b1243201c5eb3ace0

    Signed-off-by: Nikhil Nanal <nikhil.nanal@intel.com>
```

shijin-aws commented:
Sorry, my page was not refreshed. This CI is still running.
